### ISSUE TITLE
Autopilot unable to capture ILS-LOC button not working

### ISSUE THREAD

**Issue by hobblob** [FGMEMBERS/A330-200#1]
When the LOC button is pressed the plane start swinging from left to right and the heading knob on the main autopilot panel begins to turn left and then right.

**Comment by FGDATA** [comment]
It works (enganges) for me. But I found the Nav1 to switch away of the selected frequency by itself (because I do not use the FGMC)

**Comment by jan-hudec** [comment]
> When the LOC button is pressed the plane start swinging from left to right and the heading knob on the main autopilot panel begins to turn left and then right.

From that description it sounds like a poorly tuned controller. How far off course, centreline and threshold did you engage the LOC mode?

> But I found the Nav1 to switch away of the selected frequency by itself (because I do not use the FGMC).

That's strange; I don't think the radio should auto-switch when you _don't_ have active route. It might be another bug, but we should confirm it with some description of the real thing before we change it.

**Comment by JWocky** [comment]
Which airport was it? I just had an ILS failure at Kuwait, yesterday one in Phoenix and I heard form others, they had ILS problems in other airports with a variety of planes. It's maybe not Airbus specific.

**Comment by jan-hudec** [comment]
On Sunday 13 of September 2015 08:51:12 JWocky wrote:

> Which airport was it? I just had an ILS failure at Kuwait, yesterday one in
> Phoenix and I heard form others, they had ILS problems in other airports
> with a variety of planes. It's maybe not Airbus specific.

Then it would be good to also mention the version of FlightGear.

On the other hand, can you describe what that “ILS failure” looked like? It 
would likely have similar symptoms if it was related.

**Comment by JWocky** [comment]
747-8VVIP didn't catch the ILS and turned North. Under FG 3.4. Map doesn't show this light blue of a pick up either. Looking in the properties, nav showed correct frequency but no pickup.

**Comment by hobblob** [comment]
I engage the LOC button when on the map I am inside that blue arrow. Also Nav1 for some reason when you disengage the LOC you see that it has changed the CRS to a few degrees extra. This is on A330-300 on flightgear 3.4

**Comment by jan-hudec** [comment]
On Sunday 13 of September 2015 09:12:51 JWocky wrote:

> 747-8VVIP didn't catch the ILS and turned North. Under FG 3.4. Map doesn't
> show this light blue of a pick up either. Looking in the properties, nav
> showed correct frequency but no pickup.

@JWocky, that sounds like a different problem. Because the hobblob's report 
sounds like it _is_ receiving the ILS but the heading controller overshoots 
and oscillates. Which would depend on the angle at which the ILS is 
intercepted and the distance from the threshold.

But you say it didn't pick the signal at all. Since it is with stable FG, it 
would have to be with scenery. Did the map at least show the dark blue line of 
ILS with the correct frequency?

**Comment by hobblob** [comment]
I am engaging the locater at between 3500 and 300 feet and i'm always about 18 miles out maybe a bit more

**Comment by jan-hudec** [comment]
Thanks, @hobblob.

> This is on A330-300 on flightgear 3.4.

Did you say A330-**3**00? Because we have two, separate, repositories here, A330-200 and A330-300 and this is the A330-**2**00 one.

Also, please mention the _exact_ model name. There are several variants and there may be some difference that breaks AP in one of them only.

**Comment by hobblob** [comment]
sorry I meant A330-200

**Comment by hobblob** [comment]
this applies to all the variants

**Comment by hobblob** [comment]
P.S. This also happens in the a320 neo but not in the A340

**Comment by hobblob** [comment]
I was just verifying this problem on the neo and when the localizer went crazy I noticed the EICAS display was saying simulator bug. I don't know if this could be of any significance but I thought I would point it out.

**Comment by FGDATA** [comment]
Seems like something to look for after 
Thanks hobblob

**Comment by jan-hudec** [comment]
@hobblob, when the EICAS says simulator bug, there will be some messages in the console/log wingow that are very useful for tracking down what happened. Please, capture them and post them next time you encounter it.

I've recently added the "simulator bug" master warning to A320neo and didn't yet copy it to the A330 (neither -200 nor -300), so if the same problem happens with A330 (and it will, because the code is otherwise almost the same), the autopilot will often just lock up with no indication what happened. The messages on the console will be there too though.

**Comment by hobblob** [comment]
I have just run an ISL test landing with console running and it reports a nasal error "0" and then it mentioned something about "out of bounds" I think it said a string.

**Comment by legoboyvdlp** [comment]
Hey, there, theres another simulator bug here
https://github.com/FGMEMBERS/A320neo/issues/10
I tested, worked totally fine with autoland at SVMI, CAT I ILS

**Comment by hobblob** [comment]
I think this is a general issue with the whole airbus family except for the A340

**Comment by mrgooglinho** [comment]
We have 3 failure logo-wings-and lightmap lightmap and landing-lights

**Comment by legoboyvdlp** [comment]
@mrgooglinho 
I don't quite understand, say again?

@hobblob, did you check the ILS pushbutton, beside the FD, was pushed?

**Comment by hobblob** [comment]
@legoboyvdlp yes it was pushed

**Comment by hobblob** [comment]
I agree with @mrgooglinho the online aircraft with a proper working autopilot is the A340

**Comment by legoboyvdlp** [comment]
?! The A330 and 20 autopilots are much better than the 340.... if you know how to use it!

**Comment by hobblob** [comment]
I can't for some reason get the FMC to load the waypoints on SID STAR etc. It knows those routes are available. However, it displays them as empty. What could I be doing wrong? What file are the routes supposed to be stored in?

cheers

**Comment by legoboyvdlp** [comment]
Supposed to be in Database/Procedures/

**Comment by hobblob** [comment]
they still won't display on the FMC
